presto-main/src/main/java/com/facebook/presto/operator/aggregation/DoubleSumAggregation.java

```diff
@@ -30,27 +31,34 @@
     private DoubleSumAggregation() {}
 
     @InputFunction
-    public static void sum(@AggregationState NullableDoubleState state, @SqlType(StandardTypes.DOUBLE) double value)
+    public static void sum(@AggregationState LongDoubleState state, @SqlType(StandardTypes.DOUBLE) double value)
     {
-        state.setNull(false);
-        state.setDouble(state.getDouble() + value);
+        state.setFirst(state.getFirst() + 1);
+        state.setSecond(state.getSecond() + value);
     }
 
-    @CombineFunction
-    public static void combine(@AggregationState NullableDoubleState state, @AggregationState NullableDoubleState otherState)
+    @RemoveInputFunction
+    public static void removeInput(@AggregationState LongDoubleState state, @SqlType(StandardTypes.DOUBLE) double value)
```

Comment:
I don't see any tests for these. Could we add them to `TestDoubleSumAggregation` and `TestLongSumAggregation`? Or as window function queries?

Reply:
I'm not at all sure that the second commit, which adds removeInput for some Sum aggregations, is a good idea.  It increases the size of the aggregation state, which will hurt straightforward grouping queries that don't even use window functions.  There are several other aggregations, however, that can support rolling without any additional costs:  I attempted to triage the aggregations here: https://github.com/prestodb/presto/issues/8982

I think for testing, it makes sense to extend AbstractTestAggregationFunction to use removeInput() if it exists.  I can look into doing this next week.  Please keep in mind that I have only one week left at Teradata -- the 22nd will be my last day.

Reply:
I added AbstractTestAggregationFunction.testSlidingWindow in the latest version of this PR.